service/task.go

```diff
@@ -390,134 +384,35 @@ func getTaskDependencies(t *task.Task) ([]uiDep, string, error) {
 	if err != nil {
 		return nil, "", err
 	}
-
-	idToUiDep := make(map[string]uiDep)
-	// match each task with its dependency requirements
-	for _, depTask := range dependencies {
-		for _, dep := range t.DependsOn {
-			if dep.TaskId == depTask.Id {
-				idToUiDep[depTask.Id] = uiDep{
-					Id:             depTask.Id,
-					Name:           depTask.DisplayName,
-					Status:         depTask.Status,
-					RequiredStatus: dep.Status,
-					Activated:      depTask.Activated,
-					BuildVariant:   depTask.BuildVariant,
-					Details:        depTask.Details,
-					//TODO EVG-614: add "Recursive: dep.Recursive," once Task.DependsOn includes all recursive dependencies
-				}
-			}
-		}
-	}
-
-	idToDep := make(map[string]task.Task)
-	for _, dep := range dependencies {
-		idToDep[dep.Id] = dep
+	taskMap := map[string]*task.Task{}
+	for _, t := range dependencies {
+		taskMap[t.Id] = &t
```

Comment:
See my other comment about the range operator and addresses elsewhere.

Reply:
fixed